config.py

```diff
@@ -120,6 +123,32 @@ def __init__(self) -> None:
             logging.info("nocrawl_mode is True, using nocrawl_path_to_audit_log")
             self.config["max_links_per_domain"] = 1
 
+    def __automatically_configure(self) -> None:
```

Comment:
note: I'm open to alternative names for this method

I'm on the fence as I feel like it would be better if it was clearer on what was being automatically configured, but I can't think of a better name that isn't really long...

Reply:
`__do_default_config()` ? 🤷🏻‍♂️ 
I have no problem with functions that start with 'do', indicating they have some 'special job'.

Reply:
I've gone with `__resolve_automatic_settings`

Reply:
that assumes though that all the `auto` settings will be OS-based, which is currently true but might not always (in particular, I'm planning to explore inferring the thread number based on available CPUs)

Reply:
yup all options are/will still be configurable - the idea is that options supporting automatic configuration do so when their value is set to `auto`, so for the threading the logic is something like this:

```
        if self.config["thread_count"] == "auto":
            self.thread_count = len(os.sched_getaffinity(0))
```